pages/datasets/[datasetid]/access/request/[requestid].vue

```diff
@@ -11,21 +14,22 @@ const { datasetid, requestid } = route.params as {
 };
 
 const { data: req, error } = await useDownloadRequest(requestid);
+
 const { data: dataset } = await useDataset(datasetid);
 const { userDetails } = await useUserDetails();
 
 const user = userDetails.value;
 const request = req.value as any;
 
-const userKey = user.email.replace(/[^a-zA-Z0-9]/g, "");
-const [username, host] = request?.download_uri?.split("@");
-
-const notUserDownload = user.id !== request.user_details_id;
+const notUserDownload = user?.id !== request.user_details_id;
 const dataReady = request.status === "READY";
 const isExpired = request.status === "EXPIRED";
 
-const privateKeyUrl = `/api/downloads/key/${requestid}`;
-const privateKeyFilename = `${userKey}.txt`;
+const userKey = user?.email.replace(/[^a-zA-Z0-9]/g, "");
+const sasUriUrl = `/api/downloads/sasuri/${requestid}`;
```

Comment:
Is there a corresponding change for this `sasuri`?

I don't think it will behave quite the same as we've done with the SSH key as the URI is already returned from the `useDownloadRequest` via the `download_uri` property of the `req` object.

Reply:
@kirenotneb this was mostly just a placeholder we could use to quickly slot in the SAS URI logic, but would it be more appropriate to rebase this change onto the branch where that code change was made and to merge it there? If not I can update this

Reply:
I think the point that Erik is trying to make is that we had to implement a TypeScript file for the API endpoint used to download the SSH key (it's at `server/api/downloads/key/[requestid].get.ts`). This PR doesn't include server-side TypeScript for this `/api/downloads/sasuri/${requestid}` endpoint.

Also, if this is truly just a URI, I would probably display it in the UI instead of using a download. You could easily include a button to facilitate copying the URL to the clipboard. For an example, a previous iteration of this file used to copy the SSH key to the clipboard (see ce280a1).

Reply:
fair point, that was a misunderstanding on my part. if you'll permit me a little time I'll add that to this branch and PR.